main.py

```diff
@@ -84,43 +87,50 @@ def shutdown(self):
         If you have some cleanup procedure, insert it here.
         """
 
-    @rest('/v2/evc/', methods=['GET'])
+    @rest("/v2/evc/", methods=["GET"])
     def list_circuits(self):
         """Endpoint to return circuits stored.
 
         If archived is set to True return all circuits, else only the ones
         not archived.
         """
-        log.debug('list_circuits /v2/evc')
-        archived = request.args.get('archived', False)
+        log.debug("list_circuits /v2/evc")
+        archived = request.args.get("archived", False)
         circuits = self.storehouse.get_data()
         if not circuits:
             return jsonify({}), 200
         if archived:
             return jsonify(circuits), 200
-        return jsonify({circuit_id: circuit
-                        for circuit_id, circuit in circuits.items()
-                        if not circuit.get('archived', False)}), 200
-
-    @rest('/v2/evc/<circuit_id>', methods=['GET'])
+        return (
+            jsonify(
+                {
+                    circuit_id: circuit
+                    for circuit_id, circuit in circuits.items()
+                    if not circuit.get("archived", False)
+                }
+            ),
+            200,
+        )
+
+    @rest("/v2/evc/<circuit_id>", methods=["GET"])
     def get_circuit(self, circuit_id):
         """Endpoint to return a circuit based on id."""
-        log.debug('get_circuit /v2/evc/%s', circuit_id)
+        log.debug("get_circuit /v2/evc/%s", circuit_id)
         circuits = self.storehouse.get_data()
 
         try:
             result = circuits[circuit_id]
         except KeyError:
-            result = f'circuit_id {circuit_id} not found'
-            log.debug('get_circuit result %s %s', result, 404)
-            raise NotFound(result)
-
+            result = f"circuit_id {circuit_id} not found"
+            log.debug("get_circuit result %s %s", result, 404)
+            raise BadRequest(result) from KeyError
         status = 200
-        log.debug('get_circuit result %s %s', result, status)
+        log.debug("get_circuit result %s %s", result, status)
         return jsonify(result), status
 
-    @rest('/v2/evc/', methods=['POST'])
-    def create_circuit(self):
+    @rest("/v2/evc/", methods=["POST"])
+    @validate(spec)
```

Comment:
I've just realized we still have some of the old validation in place for `EVCBase`, are we going to remove part of it or most of it? Like this part here https://github.com/ajoaoff/mef_eline/blob/oas-validation/models/evc.py#L187-L205? Maybe this could still be useful when not using this in an endpoint, but might be related to another question that I asked regarding validations of dicts/objects without a request below.